tools/Az.Tools.Predictor/Az.Tools.Predictor/AzPredictor.cs

```diff
@@ -83,11 +83,22 @@ public void StartEarlyProcessing(IReadOnlyList<string> history)
 
             if (history.Count > 0)
             {
-                if (_lastTwoMaskedCommands.Any())
-                {
-                    _lastTwoMaskedCommands.Dequeue();
-                }
-                else
+                // We try to find the commands to request predictions for.
+                // We should only have "start_of_snippet" when there are no enough Az commands for prediction.
+                // We then ignore that when there are new "start_of_snippet".
+                // This is the scenario.
+                // 1. New-AzResourceGroup -Name ****
+                // 2. $resourceName="Test"
+                // 3. $resourceLocation="westus2"
+                // 4. New-AzVM -Name $resourceName -Location $resourceLocation
+                //
+                // We'll replace 2 and 3 with "start_of_snippet" but if we request prediction for 2 and 3, that'll reset the
```

Comment:
Just wanted to say these comments are very helpful being new to the code base. I know the logic of this application can get complicated at times. Just wanted to say we should look for clarification comments like this when we can add them :)

Reply:
Agreed!